ui/src/app/components/rename-modal.spec.tsx

```diff
@@ -60,3 +63,70 @@ describe('should show error if new name already exist', () => {
     }
   );
 });
+
+describe('RenameModal', () => {
+  let renameModal: RenameModal;
+
+  beforeEach(() => {
+    renameModal = new RenameModal({
+      existingNames: [],
+      oldName: '',
+      onCancel: () => {},
+      onRename: () => {},
+      resourceType: ResourceType.NOTEBOOK,
+    });
+  });
+
+  describe('generateNewName', () => {
+    it('should append suffix based on oldName format for notebooks', () => {
+      const newName = renameModal.generateNewNameForValidation(
+        'newName',
+        'oldName.RMD',
+        ResourceType.NOTEBOOK
+      );
+      expect(newName).toEqual('newname.rmd');
```

Comment:
I don't think this is what we want, because .Rmd is the canonical form.  We would be forcing these files into a non-canonical form with every rename.